File: typescript/src/renderer.ts
import { Rectangle } from "./shapes/rectangle.js";

function newRenderer(someShape: Shape) {
    return {
        draw() {
            const area: number = someShape.computeArea()
            console.log("Shape drawn\n" + "Its area is " + area)
        }
    }
}

export { newRenderer }

File: java/src/main/java/edu/cmu/cs/cs214/lab02/Renderer.java
package edu.cmu.cs.cs214.lab02;

import edu.cmu.cs.cs214.lab02.shapes.*;

public class Renderer {
    private Shape someShape;
    
    Renderer(Shape someShape) {
        this.someShape = someShape;
    }

    void draw() {
        double area = someShape.getArea();

        // assume implementation

        System.out.println("Shape printed\n" + "Its area is " + area);
    }
}
